flash/image/detection/model.py

```diff
@@ -76,132 +52,38 @@ class ObjectDetector(Task):
 
     """
 
-    backbones: FlashRegistry = OBJ_DETECTION_BACKBONES
+    heads: FlashRegistry = OBJECT_DETECTION_HEADS
 
     required_extras: str = "image"
 
     def __init__(
         self,
         num_classes: int,
-        model: str = "fasterrcnn",
-        backbone: Optional[str] = None,
-        fpn: bool = True,
+        backbone: Optional[str] = "resnet18_fpn",
+        head: Optional[str] = "retinanet",
         pretrained: bool = True,
-        pretrained_backbone: bool = True,
-        trainable_backbone_layers: int = 3,
-        anchor_generator: Optional[Type['AnchorGenerator']] = None,
-        loss=None,
-        metrics: Union[Callable, nn.Module, Mapping, Sequence, None] = None,
+        metrics: Optional['IceVisionMetric'] = None,
         optimizer: Type[Optimizer] = torch.optim.AdamW,
-        learning_rate: float = 1e-3,
+        learning_rate: float = 5e-4,
         serializer: Optional[Union[Serializer, Mapping[str, Serializer]]] = None,
+        image_size: Optional[int] = None,
         **kwargs: Any,
     ):
         self.save_hyperparameters()
 
-        if model in _models:
-            model = ObjectDetector.get_model(
-                model, num_classes, backbone, fpn, pretrained, pretrained_backbone, trainable_backbone_layers,
-                anchor_generator, **kwargs
-            )
-        else:
-            ValueError(f"{model} is not supported yet.")
-
         super().__init__(
-            model=model,
-            loss_fn=loss,
-            metrics=metrics,
+            num_classes=num_classes,
+            backbone=backbone,
+            head=head,
+            pretrained=pretrained,
+            metrics=metrics or [SimpleCOCOMetric(COCOMetricType.bbox)],
+            image_size=image_size,
             learning_rate=learning_rate,
             optimizer=optimizer,
             serializer=serializer or DetectionLabels(),
+            **kwargs,
         )
 
-    @staticmethod
-    def get_model(
-        model_name,
-        num_classes,
-        backbone,
-        fpn,
-        pretrained,
-        pretrained_backbone,
-        trainable_backbone_layers,
-        anchor_generator,
-        **kwargs,
-    ):
```

Comment:
Could we get a torchvision adapter instead of deleting it. Some users might rely on this. This is a BC.

Reply:
The old torchvision stuff wasn't really working. All the models from before are now available in the icevision integration